src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -184,19 +183,24 @@ private bool DoShutdown(
                 // When an OS shutdown request is received, try to close all non-owned windows. Windows can cancel
                 // shutdown by setting e.Cancel = true in the Closing event. Owned windows will be shutdown by their
                 // owners.
-                foreach (var w in Windows.ToArray())
+                foreach (var w in new List<Window>(_windows))
```

Comment:
Nit: why the extra `List`? Not that the extra single allocation will matter in this case, but it also seems unnecessary compared to the array used previously.

Reply:
Just a minor change to remove Linq usage. As we need a copy of the windows collection.